src/app/components/import-annotations/widgets/annotation-import-issue.component.html

```diff
@@ -0,0 +1,104 @@
+<h5>
+  <fa-icon class="pe-1" [icon]="['fas', 'circle-info']"></fa-icon>Instructions
+</h5>
+
+<p>
+  You can use this interface to upload annotations from human or machine
+  generated datasets.
+</p>
+
+@if (hasUncommittedFiles()) {
+  <div class="uncommitted-file-warning alert alert-warning">
+    <h6>
+      <fa-icon class="pe-1" [icon]="['fas', 'triangle-exclamation']"></fa-icon>
+      Warning
+    </h6>
+
+    <strong>Your files have not been saved</strong>. Please review your uploaded
+    files, and click "Import Annotations" to save your annotations.
+  </div>
+}
+
+@if (annotationImport.importErrors().length > 0) {
+  <section class="error-section widget-section p-3 border-top">
+    <h5>How to handle errors</h5>
+
+    <div class="error-cards">
+      @if (hasEventErrors()) {
+        <div class="widget-card">
+          <h6 class="callout callout-danger">Event errors</h6>
+
+          <p>
+            Event errors are broken down per-row in the "Identified Events"
+            table.
+          </p>
+
+          <p>
+            Errors can arise due to invalid values, incorrect data types, or
+            referencing audio recordings that you don't have permission to
+            access.
```

Comment:
This bit is a bit redundant.

```suggestion
            You may have to change your files to correct
            the errors so we can understand your data.
```

Reply:
I like the idea of reducing the amount of text shown.

Although I'm a bit concerned about being vague.

---

In my mind, these error widgets are attempting to assist users in fixing their data.

e.g. Note that if you don't have access to an audio recording, the error that you will receive will be "audio recording not found", which doesn't provide much help with fixing the error.

---

After writing this out, I think the root cause is that some of the errors returned by the api are a bit vague.

E.g. we should return different error messages for unauthorized access and incorrect data types (e.g. strings instead of `int`'s)

Reply:
>e.g. Note that if you don't have access to an audio recording, the error that you will receive will be "audio recording not found", which doesn't provide much help with fixing the error.

It _should_ actually tell you don't have permission - i specifically made the error distinct.

> Although I'm a bit concerned about being vague.

Fair, but then we should provide better error messages, customised to the errors we're seeing.

A basket list of possible errors is just text people don't want to read

Reply:
![image](https://github.com/user-attachments/assets/28c84693-cf5e-4f7f-b73f-05021cc466ac)

You're correct, the permission failure is distinct per event returned.

However, the errors shown in the sidebar are using the errors returned from the response's `meta.error.info` object.

![image](https://github.com/user-attachments/assets/7bbfbaf8-b354-4b5e-a7d7-f78c05428ab3)

In this case, the error message for the file is "Validation failed".

Reply:
Yeah and?

You have a function for searching through to get error types